### DISCUSSION TITLE
Support Screen Share feature for haishin_kit

### DISCUSSION THREAD

**Discussion by cade3489** [HaishinKit/HaishinKit.swift#1572] [Idea]
I am currently using the haishin_kit: ^0.13.0 package for a Flutter project and was wondering if there are any plans to add a screen-sharing feature in the near future.

Screen sharing would be a valuable addition to the package, especially for apps involving live streaming of presentations, tutorials, or gameplay. If this feature is on the roadmap, I would love to know more about its expected release timeline.

Thank you for your time and for maintaining such a great package! I look forward to your response.

**Comment by shogo4405** [comment]
There are no plans for this. Additionally, there are no plans for screen sharing code as there is no benefit to unifying the code between iOS and Android.

- For iOS, it's built in a dedicated execution environment called Broadcast Upload Extension. In this case, you would prepare a SampleHandler like this. Also, since the execution environment is different from the app, it operates in a dedicated sandbox, so it's unclear whether it would work with the Dart VM.
  - https://github.com/shogo4405/HaishinKit.swift/blob/main/Examples/iOS/Screencast/SampleHandler.swift
 
- For Android, you would set up a android.app.Service like this.
   - https://github.com/shogo4405/HaishinKit.kt/blob/main/app/src/main/java/com/haishinkit/app/MediaProjectionService.kt

Screen sharing is natively supported by HaishinKit(.swift|kt), so I think it would be best to refer to this code and work on the native layer of the main application.
